### PR TITLE
Dudnik

### PR DESCRIPTION
**Pull request by Dudnik25**

### CODE REVIEW

File: src/index.html [hunk 1]
```diff
@@ -76,7 +76,7 @@ <h2 class="title__heading">
         </div>
         <ul class="about-us__advantages">
           <li class="about-us__advantage advantage">
-            <img src="#" alt="The whole world in our" class="advantage__icon">
+            <img src="images/about-us-1.png" alt="The whole world in our" class="advantage__icon">
```
Comment (DZakh) [inline src/index.html]: I usually store images like this:
`images/about-us/image-name.png`
Where `image-name` is something that describes the image in few words.

Comment (DZakh) [reply src/index.html]: It gives easier understanding what's the image is

File: src/stylesheets/components/_advantage.scss [hunk 2]
```diff
@@ -1,3 +1,25 @@
...
+  justify-content: center;
+  align-items: center;
+  text-align: center;
+
+  &__icon {
+    width: 70px;
+  }
+
+  &__heading {
+    font-size: $title-font-size;
```
Comment (DZakh) [inline src/stylesheets/components/_advantage.scss]: You cannot access the scss variable like this, because if you initialized it inside of `_variables.scss`'s map, it becomes a css-variable, so, to get it you should use either `var(--title-font-size)` ,or scss functions `v(title-font-size)`, or `v-get(title-font-size)`.
Well, but if your variant works - that's strange.

File: src/stylesheets/components/_advantage.scss [hunk 3]
```diff
@@ -1,3 +1,25 @@
...
+    font-weight: 600;
+    margin-top: 30px;
+    max-width: 260px;
+  }
+
+  &__description {
+    margin-top: 20px;
```
Comment (DZakh) [inline src/stylesheets/components/_advantage.scss]: I prefer use `padding-top` instead of `margin-top` because margin has a weird issue with overlapping. There is an article about it on Css-Tricks.

File: src/stylesheets/components/_title.scss [hunk 4]
```diff
@@ -1,3 +1,18 @@
 .title {
-  
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  align-items: center;
```
Comment (DZakh) [inline src/stylesheets/components/_title.scss]: Hmm, I don't understand what's the purpose of the stuff in this situation. I don't need it. Replace with `margin: 0 auto;`.

File: src/index.html [hunk 5]
```diff
@@ -117,7 +117,7 @@ <h2 class="title__heading">
         </div>
         <ul class="our-services__list">
           <li class="our-services__service service">
-            <img src="#" alt="Research" class="service__icon">
+            <img src="images/smartphone.png" alt="Research" class="service__icon">
```
Comment (DZakh) [inline src/index.html]: The some thing as in the first comment, but here you did it better. My way to store images:
`images/our-services/smartphone.png`
For this web page it may be too much, but when you have a bunch of images it makes easier to manipulate with them.

File: src/stylesheets/components/_advantage.scss [hunk 6]
```diff
@@ -1,5 +1,3 @@
-$title-font-size: 20px;
-
 .advantage {
   display: flex;
   flex-direction: column;
```
Comment (DZakh) [inline src/stylesheets/components/_advantage.scss]: I don't see any sense in this `flex`.

File: src/stylesheets/components/_service.scss [hunk 7]
```diff
@@ -1,3 +1,46 @@
...
+  padding: 50px 10px;
+  background-color: v(bg-clr);
+
+  &:hover {
+    background-color: $hover-bg-color;
+    color: $hover-text-color;
+    box-shadow: 0px 5px 20px -10px black;
+    & .service__read-more {
+      border: 2px solid $hover-text-color;
+    }
```
Comment (DZakh) [inline src/stylesheets/components/_service.scss]: It's better to use css-variables for colors. Doing like this we may super easily switch colors with JS. It may be helpful if we want to add something like Dark Mode, and also it structures all colors in one place.

File: src/stylesheets/components/_service.scss [hunk 8]
```diff
@@ -1,3 +1,46 @@
...
+  }
+
+  &__icon {
+    width: 60px;
+    margin-bottom: 30px;
+  }
+
+  &__heading {
+    max-width: 270px;
+    font-size: v(heading-font-size);
```
Comment (DZakh) [inline src/stylesheets/components/_service.scss]: Note (there is no any mistake): talking about other things beyond of colors, it's not as useful to use css-variables for them, but there is also no something bad.

File: src/index.html [hunk 9]
```diff
@@ -204,37 +204,73 @@ <h2 class="title__heading">
...
-            <li class="projects__filter-item">
+            <li class="projects__filter-item" data-cat="design">
               Web design
             </li>
-            <li class="projects__filter-item">
+            <li class="projects__filter-item" data-cat="uiux">
               UI/UX design
             </li>
-            <li class="projects__filter-item">
+            <li class="projects__filter-item" data-cat="mockups">
```
Comment (DZakh) [inline src/index.html]: cat? :-D

File: src/index.html [hunk 10]
```diff
@@ -204,37 +204,73 @@ <h2 class="title__heading">
...
             </li>
-            <li class="projects__project project">
-              <!-- Figure it out by your-self -->
+            <li class="projects__project project" data-cat="mockups">
+              <a href="#" class="project__link">
+                <img src="images/projects/project-6.jpg" alt="Project Name" class="project__image">
+                <div class="project__hover">
+                  <h3 class="project__heading">Hand Watch</h3>
+                  <p class="project__description">ui/ux design</p>
+                </div>
```
Comment (DZakh) [inline src/index.html]: It's better to describe what's the block about instead what it does. I understand that it has hover effect, but in reality it's something like `project__title-wrap`... See, with your class name I cannot even understand what the block is.

Comment (DZakh) [reply src/index.html]: The same problem with the `project__link`.

File: src/index.js [hunk 11]
```diff
@@ -36,4 +37,39 @@ function link(e) {
     //Links code
 }
 
+// -----Projects Section-----
+const projectsButton = document.querySelectorAll('.projects__filter-item');
+const projects = document.querySelectorAll('.projects__project');
+
+[...projectsButton].forEach(function (element) {
+    element.addEventListener('click', portfolio);
+});
```
Comment (DZakh) [inline src/index.js]: Let's use ES6. 
```js
[...projectsButton].forEach(e => {
    e.addEventListener('click', portfolio);
});
```

File: src/index.js [hunk 12]
```diff
@@ -36,4 +37,39 @@ function link(e) {
...
+            if (element.dataset.cat === e.target.dataset.cat) {
+                element.classList.remove('projects__project_hide');
+            } else {
+                element.classList.add('projects__project_hide');
+            }
+        });
+    }
+}
+
+function portfolioButtonActive(element) {
```
Comment (DZakh) [inline src/index.js]: Maybe `portfolioButtonActivate(e)`?

File: src/stylesheets/abstracts/_variables.scss [hunk 13]
```diff
@@ -17,9 +17,11 @@ $v-globals: (
   
   /// Colors
   text-clr: #212121,
+  hover-text-clr: #fff,
```
Comment (DZakh) [inline src/stylesheets/abstracts/_variables.scss]: We already have `bg-clr`.

File: src/stylesheets/components/_project.scss [hunk 14]
```diff
@@ -1,3 +1,40 @@
...
+
+  &__hover {
+    position: absolute;
+    top: 0; bottom: 0; left: 0; right: 0;
+    background-color: rgba(0, 0, 0, .6);
+    font-weight: 600;
+    display: flex;
+    flex-direction: column;
+    justify-content: center;
+    align-items: center;
```
Comment (DZakh) [inline src/stylesheets/components/_project.scss]: As always, what do you want to get with this?

File: src/stylesheets/abstracts/_variables.scss (hunk 13 above)
Comment (Dudnik25) [reply src/stylesheets/abstracts/_variables.scss]: yes, but I think it's better to be able to separately change the color of the text for the hover

File: src/stylesheets/components/_project.scss (hunk 14 above)
Comment (Dudnik25) [reply src/stylesheets/components/_project.scss]: This is good for aligning multiple elements vertically

File: src/stylesheets/abstracts/_variables.scss (hunk 13 above)
Comment (DZakh) [reply src/stylesheets/abstracts/_variables.scss]: Ok, it makes sense

File: src/index.js (hunk 11 above)
Comment (Dudnik25) [reply src/index.js]: I'm not sure "for of" will work correctly everywhere

File: src/stylesheets/components/_advantage.scss (hunk 2 above)
Comment (Dudnik25) [reply src/stylesheets/components/_advantage.scss]: I was guided by your comments
// Use v(bg-color) to get var(--bg-color)
// Use v-get(bg-color) to get $bg-color

File: src/index.js (hunk 11 above)
Comment (DZakh) [reply src/index.js]: I didn't talk about `for of`, `forEach` is totally fine. Look at my example above.

File: src/stylesheets/components/_project.scss (hunk 14 above)
Comment (DZakh) [reply src/stylesheets/components/_project.scss]: But they are vertical from the most beginning. Why not to use `paddings` and `margin: 0 auto;`?

File: src/index.js [hunk 15]
```diff
@@ -134,5 +134,22 @@ function resizeEvent() {
     resizeSlider();
 }
 
+// -----Instagram Photo-----
+async function getPhoto() {
+    const article = document.querySelectorAll('.article');
+    const data = await fetch('https://www.instagram.com/website_of_the_day/?__a=1')
+        .then(response => response.json())
+        .then(data => data.graphql.user.edge_owner_to_timeline_media.edges);
```
Comment (DZakh) [inline src/index.js]: There may be an Ajax class somewhere in the boilerplate. It has kind of error catcher and also simple to use.
With it it'd look something like this:
```
const data = await Ajax.get('https://www.instagram.com/website_of_the_day/?__a=1');  
/* I'm not sure about the name */  
const post = data.graphql.user.edge_owner_to_timeline_media.edges;  
```

File: src/stylesheets/components/_article.scss [hunk 16]
```diff
@@ -0,0 +1,112 @@
...
+  &__background {
+    position: absolute;
+    top: 0;
+    right: 0;
+    bottom: 0;
+    left: 0;
+    width: 100%;
+    height: 100%;
+    object-fit: cover;
+  }
```
Comment (DZakh) [inline src/stylesheets/components/_article.scss]: As I remember there is a mixin for it in the boilerplate.

File: src/index.js (hunk 12 above)
Comment (fabysshk) [reply src/index.js]: classList.toggle?

File: src/index.js [hunk 17]
```diff
@@ -72,4 +72,67 @@ function portfolioButtonActive(element) {
     });
 }
 
+// -----Slider-----
+const slider = document.querySelector('.members__list');
+const sliderItemsCount = slider.querySelectorAll('.members__person').length;
```
[the remark this answers is not in the document]
Comment (fabysshk) [reply src/index.js]: Take length after
const items=document.querySelectorAll(".item")

function(){
const itemsLength=items.length}

File: src/index.js [hunk 18]
```diff
@@ -72,4 +72,67 @@ function portfolioButtonActive(element) {
...
+const sliderDots = document.querySelector('.members__slider-dots');
+let sliderItemWidth = slider.querySelector('.members__person').offsetWidth;
+let sliderBtnCount = Math.ceil(sliderItemsCount / (slider.offsetWidth / sliderItemWidth));
```
[the remark this answers is not in the document]
Comment (fabysshk) [reply src/index.js]: https://developer.mozilla.org/ru/docs/Web/API/Element/getBoundingClientRect

File: package.json [hunk 19]
```diff
@@ -12,6 +12,7 @@
   "license": "MIT",
   "dependencies": {
     "@babel/polyfill": "^7.6.0",
+    "fonts": "0.0.2",
```
Comment (DZakh) [inline package.json]: What is it? I don't need any dependency for Icomoon.
